tactics/tactics.ml

```diff
@@ -1770,6 +1770,13 @@ let descend_in_conjunctions avoid tac (err, info) c =
 (*            Resolution tactics                    *)
 (****************************************************)
 
+let tclORELSEOPT t k =
+  Proofview.tclORELSE t
+    (fun (e, info) ->
+    match e with
+    | PretypeError _ | RefinerError _ | UserError _ | NotEnoughProducts | Failure _ -> k (e, info)
```

Comment:
Why these particular errors?  Is this matching the behavior of some other function?

Reply:
I copy-pasted what was done before and added the new `NotEnoughPremisses` (`NotEnoughProducts` was a bad autocompletion of emacs). The alternative would be to use `noncritical` (see also the question I asked at [#11711](https://github.com/coq/coq/pull/11711#issuecomment-908508403)).

Reply:
I think you should use noncritical here, unless the commit message introducing the filtering has a particular justification for it.  Does the use of the filter vs noncritical impact the bench?  (Maybe it's worth temporality sticking an `assert false` on the case where the filter doesn't overlap with noncritical just to see if there's anything on the CI that can distinguish between them?  And if so, we can minimize it and add it to the test-suite to exercise this code path, and then decide what to do there with more information?)